### PR TITLE
WP Migration: Add tools/migration menu

### PR DESCRIPTION
**Pull request by jonathansadowski**
This change is behind a feature-flag and is part of an in-progress larger project. For context, see pbkcP4-8-p2.

#### Changes proposed in this Pull Request

* Adds a feature-flag (`tools/migrate`)
* Adds "Migrate" under the "Tools" menu in the sidebar
* Adds `/migrate` route
* Adds new page (`SectionMigrate`) at new route
* Adds `SiteSelector` for selection of Jetpack sites on new page

#### Testing instructions

* Check out this pull request and run Calypso locally, or view it on calypso.live with the flag `tools/migrate`.
* Viewing `calypso.localhost:3000/migrate/`, you should see a `SiteSelector` component allowing you to choose a site to migrate a Jetpack site into.
* When you select a site, you should see another `SiteSelector` component, listing all the Jetpack sites your account has access to. The Tools section in the sidebar should show the new Migrate option.

### CODE REVIEW

File: client/my-sites/sidebar/index.jsx [hunk 1]
```diff
@@ -232,6 +232,30 @@ export class MySitesSidebar extends Component {
...
 
+	trackMigrateClick = () => {
+		this.trackMenuItemClick( 'migrate' );
+		this.onNavigate();
+	};
+
+	migrate() {
+		const { path, siteSuffix, translate } = this.props;
+
+		if ( ! isEnabled( 'tools/migrate' ) ) {
```
Comment (andfinally) [inline client/my-sites/sidebar/index.jsx]: Would it make sense to move this above the `const` initialisation?

**Comment by andfinally** [comment]
Maybe too early to look at in this PR, but it's just occurred to me this section should maybe only be visible in Atomic sites – or we should replace the `SiteSelector` with a placeholder on simple sites.

When I hit `/tools/migrate`, I'm able to select one of my JP sites from the site selector on that page – potentially leading to a confusing situation where I try to migrate a JP site to iteself.

**Comment by mattsherman** [comment]
> Maybe too early to look at in this PR, but it's just occurred to me this section should only be visible on Atomic sites 

Since we plan on implementing the ability to purchase a Business plan and convert to Atomic as part of the MVP, I think we are fine with this right now, as long as the actual migration fails with an error message that helps a dev/tester understand what went wrong.

File: client/my-sites/migrate/section-migrate.js [hunk 2]
```diff
@@ -0,0 +1,74 @@
...
+import { getSelectedSiteId } from 'state/ui/selectors';
+
+/**
+ * Style dependencies
+ */
+import './section-migrate.scss';
+
+class SectionMigrate extends Component {
+	state = {
+		sourceSite: null,
```
Comment (andfinally) [inline client/my-sites/migrate/section-migrate.js]: We need to change the state name to `sourceSiteId` here too.

**Comment by andfinally** [comment]
LGTM! I'm still not sure about the filtering of sites. My SA tells me I have 12 JP sites (including Atomic ones), but the `SiteSelector` adds 5 superfluous ones, including `store.a8c.com` and `jurassic.ninja`. A filter for my account's capabilities would hopefully get rid of those extra sites.

I'm also wondering if we should include Atomic sites at this point. If we do, we'll probably want to advise people at some point that there's a better way to migrate Atomic to Atomic (the one using the [create site](https://href.li/?https://mc.a8c.com/atomic/apidocs-webhost/#api-Sites-create_site) endpoint).

Also noting that when you navigate away from the page and back the state is lost. This should be OK.

**Comment by mattsherman** [comment]
> LGTM! I'm still not sure about the filtering of sites. My SA tells me I have 12 JP sites (including Atomic ones), but the `SiteSelector` adds 5 superfluous ones, including two commonly used A8C sites I'm not an admin on. A filter for my account's capabilities would hopefully get rid of those extra sites.

I'll see what the filtering looks like when I test as well. 

> 
> I'm also wondering if we should include Atomic sites at this point. If we do, we'll probably want to advise people at some point that there's a better way to migrate Atomic to Atomic (the one using the create site endpoint).

I think it is okay to include Atomic ones now, since we plan on supporting Atomic sources down the road.

We should handle the better way to migrate Atomic to Atomic transparently for users -- we'll tackle that in a future milestone (I have it on my list to backlog that).
 
> Also noting that when you navigate away from the page and back the state is lost. This should be OK.

I think that is okay for this PR. How much extra effort would be involved in having the state persist though? That is something we'll definitely want longer-term (well, we will actually want that on the "lock screen" for the site, which could be this same page, depending on the design/implementation.

**Comment by andfinally** [comment]
I guess it would make sense to think about the state on this page when we implement the locking.

File: client/my-sites/migrate/controller.js [hunk 3]
```diff
@@ -0,0 +1,15 @@
...
+ */
+import React from 'react';
+
+/**
+ * Internal Dependencies
+ */
+import SectionMigrate from 'my-sites/migrate/section-migrate';
+
+export function migrateSite( context, next ) {
+	context.primary = <SectionMigrate />;
```
Comment (mattsherman) [inline client/my-sites/migrate/controller.js]: Can we put this behind the feature gate as well? Or, do check when rendering the section and return null or something.

Currently, if you go directly to `/migrate` you see the screen, even without the flag enabled.

**Comment by jonathansadowski** [comment]
> I think that is okay for this PR. How much extra effort would be involved in having the state persist though? 

I don't think that it's worth polluting global state for something that is essentially just local. Let's leave it local for now and move it later if we have a need

**Comment by jonathansadowski** [comment]
Thanks for the reviews!

**Pull request merged**